### DISCUSSION TITLE
Feature request: combine short stationary charges into a single line/session?

### DISCUSSION THREAD

**Discussion by bikeymouse** [teslamate-org/teslamate#1630] [Ideas]
Hi, 
I'm using TWCManager to charge my car with Solar energy.  On a cloudy day this means that the charging can be started and stopped a couple times during a charging session.

This causes the Charges report in Teslamate to show a lot of individual charges:
<img width="1353" alt="charges" src="https://user-images.githubusercontent.com/34967010/118285329-9f608700-b4d1-11eb-814d-1e1dd9028253.png">

So the question is, could these be combined into 1 single line in the Charges-report and into 1 single Charge-details report, based on the fact that the car was stationary and was plugged-in all the time? 

That would make the Charges-overview much more comprehensive and also would help to see in the Charge-details report what the totals for that "single session" are.

**Comment by adriankumpf** [comment]
Feature requests belong here: [💡 Ideas](https://github.com/adriankumpf/teslamate/discussions/categories/ideas). I'll move it.

**Reply by nickjupp** [reply]
Same for me with a Zappi charger/solar.
thanks

**Comment by cwanja** [comment]
Yes!

**Comment by bfayers** [comment]
I also get this, but it's due to my charging being handled by my energy provider with slots for cheaper pricing (Octopus Intelligent Go)

I have found a way to manually merge these charges (https://gist.github.com/bbock/71fe9d4d02044858555823bc0ed3a1f3) but sometimes after doing that the Battery Health dashboard will calculate an unbelievably low figure showing something like 12% in 2000 miles, I assume its related to some weirdness with how the charges end up calculating?

In any case a feature to automatically merge 'charges' that happen at the same odometer value would be great.
